### DISCUSSION TITLE
Allow $model->relationLoaded() to check for nested relations with dot notation

### DISCUSSION THREAD

**Discussion by rcoudercher** [laravel/framework#51774] [Ideas]
I just had to check if a model had a nested relation loaded. It was in a JsonResource if that matters. I tried doing ``$this->relationLoaded('contract.partner')`` with the dot notation as I had just eager loaded those relations with an identical dot notation, before realising that wasn't possible. I think it would be a nice feature to add. What do you think ?

**Comment by wizzymore** [comment]
You can do `$this->relationLoaded('contract') && $this->contract->relationLoaded('partner')`.
If the contract is always loaded you can simply use only the second condition.

**Reply by shaedrich** [reply]
This is a workaround but not an elegant solution. Here, you have two, but what if you have three or four or more levels of nesting. One could sure debate how many levels you really need for this, but in the end, it comes down to not being chainable.

**Reply by wizzymore** [reply]
Well you can create your own helper:

```php
function isRelationLoaded($model, string $relation): bool {
    $relations = explode('.', $relation);
    $relation = array_shift($relations);
    if (!$model->relationLoaded($relation)) {
        return false;
    }
    $model = $model->{$relation};
    
    foreach ($relations as $relation) {
        if (!$model->relationLoaded($relation)) {
            return false;
        }
        $model = $model->{$relation};
    }
    
    return true;
}
```

**Reply by rcoudercher** [reply]
@wizzymore I went for something similar as your helper with a recursive function to allow for any depth check. But the point is, shouldn't this be integrated into the framework ? I find it strange that you can eager loaded nested relations with the dot notation but can't check for loaded relations with same syntaxe. It wouldn't be hard to implement.

**Reply by shaedrich** [reply]
Agreed. Maybe, you two could make a PR, or are there any issues why this hasn't been done in the first place, I'm not aware of?
